kardia/abi.go

```diff
@@ -87,13 +89,80 @@ func (ec *Client) DecodeInputData(to string, input string) (*types.FunctionCall,
 		arguments[i] = parseBytesArrayIntoString(arg)
 	}
 	return &types.FunctionCall{
-		Function:   method.String(),
-		MethodID:   "0x" + hex.EncodeToString(sig),
-		MethodName: method.Name,
-		Arguments:  arguments,
+		ContractAddress: to,
+		Function:        method.String(),
+		MethodID:        "0x" + hex.EncodeToString(sig),
+		MethodName:      method.Name,
+		Arguments:       arguments,
+		Timestamp:       time.Now().Unix(),
 	}, nil
 }
 
+// UnpackLog returns a log detail
+func (ec *Client) UnpackLog(log *types.Log) (*types.Log, error) {
+	var a *abi.ABI
+	// check if the to address is staking contract, then we search for an event in staking contract ABI
+	if ec.stakingUtil.ContractAddress.Equal(common.HexToAddress(log.ContractAddress)) {
```

Comment:
🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>ifElseChain: rewrite if-else to switch statement (gocritic)